### PR TITLE
MLeap scoring

### PR DESCRIPTION
**Pull request by ancasarb**
## What changes are proposed in this pull request?
 
Improve scoring performance by creating the DefaultLeapFrame in PandasSplitOrientedDataFrame without serializing the raw frame to json only to immediately read it back. Especially with large leap frame, (de)serialization is often much slower than the actual scoring. To do this, we need to use StructType directly instead of LeapFrameSchema and use MleapStructTypeFormat from MLeap to read the schema from json.

Other more minor change is to reuse Java DSL from MLeap to remove Scala/Java collections conversions from mlflow code base. 
 
## How is this patch tested?
 
Unit tests passing.
 
### Is this a user-facing change? 

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [x] Scoring 
- [ ] Serving
- [ ] R
- [x] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

It's my first change submitted to this project, please let me know if I've missed out anything. Thank you!

### CODE REVIEW

**Comment by ancasarb** [comment]
Closing for now, while I fix the build failures, will reopen shortly.

**Review by dbczumar — APPROVED** [review]
@ancasarb Thank you for significantly simplifying our MLeap inference logic / model server and cleaning up our Java code. I've tested this change with a local scoring server and with SageMaker; everything works as expected.

I left one comment regarding the `scala-maven-plugin`. It seems like we could remove the plugin if `MLeapSchemaReader.scala` were written in Java instead. If there are no significant drawbacks, can you make this change?

File: mlflow/java/scoring/src/main/java/org/mlflow/sagemaker/MLeapSchemaReader.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.mlflow.sagemaker
```
[the remark this answers is not in the document]
Comment (ancasarb) [reply mlflow/java/scoring/src/main/java/org/mlflow/sagemaker/MLeapSchemaReader.scala]: Unfortunately, Jackson's ObjectMapper in Java doesn't work well with Scala case classes, this is why MLeapSchemaReader is written in Scala, using Spray Json. 

I've added support for doing this in MLeap itself in this commit https://github.com/combust/mleap/pull/496/commits/9725d2cc5dabc542daf08b00f988ded458efb0f2 so once that's merged and a new version of MLeap becomes available, I'll raise another PR to remove the scala plugin. What do you think?

**Comment by dbczumar** [comment]
@ancasarb That sounds great! Thanks for the explanation. I'll merge the PR as is!

**Pull request merged**